WIP: DOC: new docs landing page + tutorial overhaul

drammock opened the pull request:
This PR implements:

- a new documentation landing page
- thematic grouping of tutorials, and
- a linear path through tutorials (determined alphabetically by tutorial filename)

Ideas are very welcome at all stages (including ideas about what is included in the tutorials roadmap... some topics may be better covered with a less narrative "how-to" example). 

### Roadmap

In the roadmap below, "finished" tutorials are checked off; tutorials that are mostly finished but have unresolved issues have "*TODO notes in italics*", and tutorials not yet started have unchecked boxes.

#### working with continuous data

- [x] Loading raw data
- [x] Querying the Raw object
- [x] Subselecting data from Raw objects
- [x] Manipulating the time domain of Raw objects
- [x] Built-in plotting methods for Raw objects
- [x] Exporting and saving data from Raw objects
- [x] Annotating continuous data
- [x] Working with events
  - [ ] event masking
  - [ ] fixed-length events
  - [ ] events → annotations → events round trip
  - [ ] importing events from non-FIFF formats

#### preprocessing

- [ ] Filtering and resampling
- [x] Interpolating bad channels
- [x] Background: projectors and projections
- [x] Working with projectors in MNE-Python *TODO: missing section on when to apply projs*
- [x] Computing projectors *TODO: has several questions in it, search for "TODO"*
- [x] Plotting the spatial pattern of projectors
- [x] Signal-space separation and Maxwell filtering *TODO sample dataset with cHPI to demo movement compensation?*
- [x] Setting the EEG reference
- [ ] Artifact removal with ICA

#### epoching

- [x] Event-related epoching, subselecting, loading, saving
- [x] Plotting epochs
- [x] Epoch rejection
- [x] Adding metadata (pandas)

#### estimating evoked activity

- [x] making evoked from epochs, loading/saving evokeds
- [x] plotting evoked
- [x] combining evokeds
- [ ] stats on evokeds

#### inverse imaging  (see #5383 when revising these)

- [ ] creating source space
- [ ] creating forward solution
- [ ] noise covariance
- [ ] making inverse operator
- [ ] making source estimate (mne, dipole, beamformer?)
- [ ] plotting source estimates
- [ ] morphing source estimates
- [ ] stats on source estimates (within subj)
- [ ] stats on source estimates (between subj)

#### new miscellaneous

- [x] Working with sensor locations

#### new "discussion" pages
*NOTE: these should not be step-by-step tutorials, they are discussions more like the "background on filtering" page*

- [ ] discussion of browsing raw data: what to look for, why it's important, etc
- [ ] background discussion of preprocessing (filter before/after epoching? when to use SSP vs ICA? etc) 

### Other TODOs

- [ ] homepage links used to go to (and expand) sections of the docs landing page... need new target for those
- [ ] flowchart on docs landing page: needs to be more general (beamformer, etc)
- [ ] flowchart on docs landing page: is docs landing page the right home for the flowchart?
- [ ] glossary is missing a number of terms
- [ ] filename of installation landing page is "getting started" (and in general, a few other filenames could use cleanup)
- [ ] remove redundancy in `plot_creating_data_structures`, `plot_objects_from_arrays`, `plot_info` tutorials (see #5982)
- [ ] good practices with `docdicts` (some times is confusing. see https://github.com/mne-tools/mne-python/pull/6451/files#r294985568)


cc #5937 

closes #5269

massich commented on doc/documentation.rst:
```diff
@@ -1,738 +1,189 @@
+customize your MNE-Python plots using any of matplotlib or mayavi's plotting
+commands. The intent is that MNE-Python will get most neuroscientists 90% of
+the way to their desired analysis goal, and other packages can get them over
+the finish line.
```
I love this part. I think we should point to places that are clear examples of the integrations like [this one using sklearn's pipeline and function transformer](https://mne-tools.github.io/dev/auto_tutorials/plot_sleep.html#multiclass-classification-workflow-using-scikit-learn) 

Maybe we should also have an example of a custom plotting theme and how to apply it. (I think @larsoner did something like this)

And we should have several of those integration examples and collect them into a common place. So maybe we should keep this paragraph tight and point to another page where we have a show off gallery of integrations.

jona-sassenhagen replied:
>I love this part. I think we should point to places that are clear examples of the integrations like this one using sklearn's pipeline and function transformer

I agree. This is how I frame MNE at workshops.

massich commented on doc/documentation.rst:
```diff
@@ -1,738 +1,189 @@
-        </div>
-        <div id="collapse_io" class="panel-collapse collapse">
-          <div class="panel-body">
+**Floating-point precision** MNE-Python performs all computation in memory
```
This seems a GOTCHA. Do we have more? maybe we need a gotcha section.

wmvanvliet commented:
This looks cool! Of course, now the tutorials need a place to go. Maybe create a new top-level menu item for them?

massich commented:
@wmvanvliet I think we need to find them a story. That's a bit what is being discussed in #5937, #5883 and other places.

massich commented:
here is the rendered version of documentaiton.html https://11328-1301584-gh.circle-artifacts.com/0/dev/documentation.html

agramfort commented:
here is how it looks:

https://11328-1301584-gh.circle-artifacts.com/0/dev/documentation.html

drammock commented:
FYI, I just discovered that `sphinxcontrib-mermaid` available in pip uses mermaid 7.1.0, which results in the class styling of the nodes not working.  With current mermaid release 8.0.0 it works as expected.  I'll see if I can push the `sphinxcontrib-mermaid` maintainer to release a new version to pip; otherwise maybe we can install the extension from github.

drammock commented:
here is [the dynamically-generated flowchart in-situ](https://11353-1301584-gh.circle-artifacts.com/0/dev/documentation.html#conceptual-flowchart-of-mne-python).  (see note above about the node styling not working; here is how it would look if using mermaid 8.0.0: http://dan.mccloy.info/data/mermaid-test.html)

agramfort commented:
to share all ideas, here is graph made by someone in neurospin based on the initial graph of Lauri.

![m_eeg flowchart_v5](https://user-images.githubusercontent.com/161052/52911903-61506a80-32aa-11e9-849b-48039e4748c1.png)

I see that adding images to illustrate what blocks correspond to can be quite valuable. Is it possible with mermaid too?

drammock commented:
@agramfort if you tie into the mermaid API directly (javascript), it is possible for the content of a node to be an image.  [Here is an example on stackoverflow](https://stackoverflow.com/a/51572474/1664024).  It also looks like if you're careful, you can put in `<img>` tags directly into the graph definition, as seen in [this thread](https://github.com/knsv/mermaid/issues/548).  Meanwhile, I've forked mermaid, fixed the version number, and am telling CircleCI to install `sphinxcontrib-mermaid` from a zipball of my fork, so the node class defs should work now.  In the coming days I'll see what I can do with the images in the flowchart.

agramfort commented:
sounds good

images are not a blocker though

>

larsoner commented:
This pull request **introduces 1 alert** and **fixes 1** when merging 3b1df7a1bf9422672ac085c309340cd765de531b into f9d19f48971d23113c2ef047df8910b6da37cd9f - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-4ad4669b194422eb2be14937796f1d06ee773635)

**new alerts:**

* 1 for Variable defined multiple times

**fixed alerts:**

* 1 for Variable defined multiple times

---

*Comment posted by [LGTM.com](https://lgtm.com)*

drammock commented:
not sure why circle failed, it's on tutorials I didn't really change (`plot_brainstorm_auditory.py` and `plot_brainstorm_phantom_ctf.py`).  Anyway, here's a first pass at the [working with continuous data](https://11546-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/index.html#working-with-continuous-data) tutorial series.

larsoner commented:
A rebase should fix it

larsoner commented on doc/conf.py:
```diff
@@ -397,6 +400,7 @@ def reset_warnings(gallery_conf, fname):
     'show_memory': True,
     'line_numbers': False,  # XXX currently (0.3.dev0) messes with style
     'within_subsection_order': FileNameSortKey,
+    'filename_pattern': r'/.*(?<!_)\.py$',  # anything that isn't *_.py
```
It looks like you got part of the way there but you'll need to rename `examples/datasets/spm_faces_dataset.py`, `examples/decoding/decoding_rsa.py`, `examples/visualization/make_report.py`, and four files under `examples/realtime/` to have trailing underscores

drammock commented:
oops, yeah, meant to do that but it slipped through the cracks. will fix today.

On February 26, 2019 7:55:59 AM AKST, Eric Larson <notifications@github.com> wrote:
>larsoner commented on this pull request.
>
>Can you rename the necessary no-plotting files and rebase?
>
>> @@ -397,6 +400,7 @@ def reset_warnings(gallery_conf, fname):
>     'show_memory': True,
>   'line_numbers': False,  # XXX currently (0.3.dev0) messes with style
>     'within_subsection_order': FileNameSortKey,
>+    'filename_pattern': r'/.*(?<!_)\.py$',  # anything that isn't
>*_.py
>
>It looks like you got part of the way there but you'll need to rename
>`examples/datasets/spm_faces_dataset.py`,
>`examples/decoding/decoding_rsa.py`,
>`examples/visualization/make_report.py`, and four files under
>`examples/realtime/` to have trailing underscores

drammock commented:
@larsoner I've changed those filenames, but there was no rebase to do (the previous force-pushed commit was after a rebase around 15 hours ago, and nothing has changed since then)

larsoner commented:
@drammock this line is failing to do its job in `.config/circle.yml:
```
FNAMES=$(git diff --name-only $CIRCLE_BRANCH $(git merge-base $CIRCLE_BRANCH upstream/master));
```
On CircleCI we try to get a list of files that have changed. This list is used to determine which examples to run, and which datasets to download (including the BST ones, which otherwise require confirmation, hence the timeouts).

The problem is that it does not work with files that have been *renamed*, which your PR does a lot of. Instead of seeing `tutorials/plot_brainstorm_phantom_elekta.py` in the list, we need to see the name it moved to, `tutorials/miscellaneous/plot_brainstorm_phantom_elekta.py`. Can you see if you can fix this `git` command to give us the correct name list? In your case $CIRCLE_BRANCH should just be your local `new-docs-landing-page`.

drammock commented:
aah, problem is that [this](https://github.com/mne-tools/mne-python/blob/master/mne/datasets/utils.py#L371-L373) isn't happening?  But why would it suddenly stop happening now; is it because the tutorial location moved from `/tutorials/` to `/tutorials/miscellaneous/`?

larsoner commented:
See comment above; the changed-names autodetection fails due to moves rather than changes, and this set of names is used to find the set of datasets to download (while passing a command-line switch to accept license terms).

drammock commented:
ah, yep.  Missed your prev. comment before I posted my supposition

larsoner commented:
This pull request **introduces 1 alert** and **fixes 1** when merging ee22568aac6c25a7e99025bfab273f0530b4a61c into f9d19f48971d23113c2ef047df8910b6da37cd9f - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-4b3cf4ea5504420127b66c6f86406f4f2bb2ecff)

**new alerts:**

* 1 for Variable defined multiple times

**fixed alerts:**

* 1 for Variable defined multiple times

---

*Comment posted by [LGTM.com](https://lgtm.com)*

drammock commented:
@larsoner I think I figured out what's still wrong with circle, but I need some help fixing it.  Here are the warnings that it's choking on:

```
/home/circleci/project/doc/auto_examples/datasets/sg_execution_times.rst:14: WARNING: Unknown target name: "spm_faces_dataset".
/home/circleci/project/doc/auto_examples/decoding/sg_execution_times.rst:20: WARNING: Unknown target name: "decoding_rsa".
/home/circleci/project/doc/auto_examples/realtime/sg_execution_times.rst:10: WARNING: Unknown target name: "ftclient_rt_average".
/home/circleci/project/doc/auto_examples/realtime/sg_execution_times.rst:12: WARNING: Unknown target name: "ftclient_rt_compute_psd".
/home/circleci/project/doc/auto_examples/realtime/sg_execution_times.rst:14: WARNING: Unknown target name: "rt_feedback_server".
/home/circleci/project/doc/auto_examples/realtime/sg_execution_times.rst:15: WARNING: Unknown target name: "rt_feedback_client".
/home/circleci/project/doc/auto_examples/visualization/sg_execution_times.rst:24: WARNING: Unknown target name: "make_report".
```

These are the files that got renamed from `filename.py` to `filename_.py` to accord with the new practice of naming not-to-be-executed examples/tutorials with a trailing underscore (instead of naming to-be-executed examples/tutorials with a leading `plot_`).  I think what's happening is that those filenames are getting written to `sg_execution_times.rst` and then when that gets parsed to make an HTML file out of it, they're getting interpreted as links instead of just text, and sphinx gets mad because those links don't have matching targets anywhere.  What I can't figure out is why this doesn't fail for [expyfun](http://labsn.github.io/expyfun/auto_examples/experiments/sg_execution_times.html) (notice `version_checking_.py` works fine there).

1. am I right about what's going wrong?
2. can you suggest an approach to fixing?

larsoner commented:
Try locally changing it to _noplot instead of just trailing underscore to test this idea. If it fixes things, it could be a SG problem that trailing underscores are not handled properly. As to why expyfun does not suffer from this, it's possible that it does not use / has not used SG master since the execution times PR was merged

larsoner commented:
https://github.com/sphinx-gallery/sphinx-gallery/pull/451

drammock commented:
@larsoner in light of your lightning-quick fix to sphinx-gallery, should I just wait for that to get merged (`.circleci/config.yml` pulls a zipball of sphinx-gallery master), or should I update the circle config to pull a zipball from `sphinx-gallery/larsoner:fix-exec`?

larsoner commented:
I'd just wait, the error does not prevent the doc from being viewable in the meantime

larsoner commented:
This pull request **introduces 1 alert** when merging 4ec96426a9be4ed63d1aabed5ab7e14d9f4b04ee into e6a1f5e6e9640d2d90349b387b1395bf842560d7 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-0b0c458b333b2bc9e83cc342d32afa921c837c7d)

**new alerts:**

* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
This pull request **introduces 2 alerts** when merging d7af7dc8dcaca06c4fad976b2d03669983c8b571 into c3e7571a9b698034cda94aae53004b55265057a9 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-b9281f3d9955bd0ab36000bc43ed08a480dce185)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

jona-sassenhagen commented on tutorials/preprocessing/working_with_sensor_locations.py:
```diff
@@ -0,0 +1,145 @@
+# ^^^^
+#
+# - mne.channels.make_eeg_layout() ?
+# - mne.channels.make_1020_channel_selections() ?
```
This function is very specific to a subset of EEG analyses, and not at the stage of preparing MNE instances, I would say. (At least that's how I intended it :)) Probably not optimal here.

drammock replied:
OK, will skip `make_1020_channel_selections` here.

jona-sassenhagen commented on tutorials/preprocessing/20_plotting_projectors.py:
```diff
@@ -0,0 +1,82 @@
+.. include:: ../../tutorial_links.inc
+
+This tutorial describes how to plot the topography (spatial pattern) of a
+projector. As usual we'll start by importing the modules we need, and loading
```
Note that most EEG folks probably don't know what projectors are :) Maybe add a few nice friendly sentences about referencing schemes here, or just tell EEG folks to skip this.

jona-sassenhagen commented on tutorials/preprocessing/05_background_projectors.py:
```diff
@@ -0,0 +1,228 @@
+
+.. include:: ../../tutorial_links.inc
+
+This tutorial provides background information on :term:`projectors <projector>`
```
Note that most EEG folks probably have *never* heard of the term "projectors"!

drammock replied:
@jona-sassenhagen what do you think should change (if anything) to address that?  should we add a sentence telling EEG-only users to skip ahead to some other tutorial?  Projectors can be used with EEG-only data...  maybe a sentence here giving the high-level takeaway ("projectors are used for..." would be enough)?

jona-sassenhagen commented on tutorials/continuous_data/20_manipulating_raw_time_domain.py:
```diff
@@ -0,0 +1,67 @@
+
+sample_data_folder = mne.datasets.sample.data_path()
+sample_data_raw_file = os.path.join(sample_data_folder, 'MEG', 'sample',
+                                    'sample_audvis_raw.fif')
```
All of this annoying wrangling and constant importing of `os` makes me wonder if we should have an extra function in datasets that just outright returns a well formatted path to the key files.

drammock replied:
which files count as "key files" besides `sample_audvis_raw.fif`?  Are you thinking something like

```Python
mne.datasets.sample.get_file('raw')  # or 'stc' or whatever
```
with a set of predefined arg values that return those key files?

jona-sassenhagen commented:
Very promising!

larsoner commented:
This pull request **introduces 3 alerts** when merging 807c94391a27a6ac028fe5e1fba8a6d85f23a4ba into 01a1aef46792d22dff39f38516268434a2497795 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-645e0c72e8a77e1c6660c31b8e90ac44ffdbbfa9)

**new alerts:**

* 2 for Unused import
* 1 for Unnecessary pass

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
This pull request **introduces 2 alerts** when merging 80ee4f460de8ed51c76b7d3daf970e35e57fff4b into 41f6f2bcd0b8e90807837e131314be81e0cd5ca6 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-ca6edf0f2d59e27e49a000272f348e5ad7c2cd5c)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

drammock commented:
latest artifacts: https://11831-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/index.html

agramfort commented:
cool ! did you put the updated roadmap somewhere? just to know how long
still is road

>

massich commented:
This one is great! really great
https://11831-1301584-gh.circle-artifacts.com/0/dev/documentation.html#

I can stretch enough how I would like to land into that documentation.html.

some things I found (most probably you already have them in your list but just in case)

Now the buttons of the main page do not point properly. see https://11831-1301584-gh.circle-artifacts.com/0/dev/index.html

Maybe we should split the Tutorials with the same structure as the buttons of the main page and point there.

In the installations instructions there's a mismatch in the file naming(the file is named getting_started) + I would let the text be already in the page not only the outline.
https://11831-1301584-gh.circle-artifacts.com/0/dev/getting_started.html

drammock commented:
@agramfort I changed the initial comment of this PR to include the roadmap with checkboxes.
@massich I've also added your suggestions to the checklist at the top of this PR.

larsoner commented:
This pull request **introduces 2 alerts** when merging b409bd02fc4869b717c672c70139bb8f8fc6d62b into 561ded7f8fa4c2c63ca8f429a2da98323d28d311 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-75755688c70fcb9578fb34045063d5a31d6ce936)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
This pull request **introduces 2 alerts** when merging e8655a63384c7f4406bfc7d8566038f73490569c into b74bb10d779dfd8144a4b6fd8fc5444f441f2141 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-0bb196d7c851606c9ad566d721c69d556bd4143e)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
This pull request **introduces 2 alerts** when merging b0e72d91ff905a1e965d3a2761cac5b47e438528 into c30e70c53cd25997a8a9831ab5c3e827dd5f2966 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-0fc350e375f8a79e7f2668ff54b8f037d9060d8d)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
This pull request **introduces 2 alerts** when merging 97551293ac9e2875dee5bc75fc066bad8f85aeca into 56909cdbf35511d9837a5999ce1ffe2f47eee7bb - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-1660dd4f22819d03679ba25813d0a4b9fa3cb331)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
This pull request **introduces 2 alerts** when merging bc917f60ed75ecafa38103bffe632a7f1f1659d5 into 027f82d79b8055bad5c997829d409035de348fe8 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-b1a6dd22a3047ba36faf4b968784edef277d45bb)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
This pull request **introduces 2 alerts** when merging 03e810d2dbec76b833a1a46128aa4027e78145e3 into aa7b3a0ba1038e6ad7fb357e60844696e234c40b - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-8971d2d53d4df8da4dbc135cf18b832c6ce7e3b1)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
This pull request **introduces 2 alerts** when merging 8af7554e4b202e01a5e4bdc2ed8d4497d6133058 into b3dfd7a1dd4cfc2fb501f9db9c4bcb3eb158cc2c - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-e2c43a231a56e1123b5a4c974291a54e3d99f177)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
```
Killed
Makefile:44: recipe for target 'html_dev' failed
make: *** [html_dev] Error 137
Exited with code 2
```
Usually this is a memory problem. Probably you are using sample, uncropped, preloaded. Crop to 60 sec (and say this is for memory reasons) and it should work.

Also you need a rebase

larsoner commented:
(You can check this with `mprof`, anything over 2GB memory usage is asking for trouble)

larsoner commented:
This pull request **introduces 2 alerts** when merging b36a8262f30b36b1f5edee99134aeca05cc22797 into 9cc5d1457c29151eea73fa7f14619724182960c1 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-94080423320f68ddd42e07c51e16c32116774485)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
This pull request **introduces 2 alerts** when merging d7fd58f3601451de10d53506a19eff60dc183a87 into 46055a3a4122bf30740f6cefcfb03310ce064738 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-dd51fe47a93f73650be5742b3dbf80d887c7ea85)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
Errors are real (examples should not emit warnings so they are treated as errors)

https://circleci.com/gh/mne-tools/mne-python/12486

larsoner commented:
This pull request **introduces 2 alerts** when merging fdb50519ef59c8fe76863e7194117ca2bbb1ffe7 into ab49f9c124fe61f0d1c437ab33c2970833f2e71e - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-bd1e7a36e621eb462230ebc9ccf94028c4b5dd01)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
This pull request **introduces 2 alerts** when merging cf953efebb147f984424cf862731e5d7238f5012 into 767cd9246370d25b132b4d98206095e15779044f - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-d73db6ba5ffcbb3a046b5d6536c60a74d3bb27d5)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
This pull request **introduces 2 alerts** when merging 27a909f4cf788b21d1083d43e6f5c0508ddb06f4 into 767cd9246370d25b132b4d98206095e15779044f - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-0b8b456f25c7e3af31d18b5938a91d1d4f69cc54)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

drammock commented:
rendered docs: https://12627-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/index.html

larsoner commented:
This pull request **introduces 2 alerts** when merging c164e414f9664d3666aafba639e1cccc424545cf into 5a9ae3c3d66b6275fffec217cf560acf651913d7 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-0a4132d630fb67978959eabfc3bc0c02cb55d20a)

**new alerts:**

* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

larsoner commented:
@drammock is this PR still valid, or should we close? Can still use the list at the top, or maybe better, move to a dedicated issue.

drammock commented:
@larsoner I've been avoiding closing it because I still reference the Circle rendering from time to time; I've been assuming that would get GC'd if we closed this PR.  Is that true?  If not, I'm fine with closing and migrating the checklist to a new "docs tracking" issue.

larsoner commented:
Apparently they keep them forever

https://discuss.circleci.com/t/circleci-artifacts-expiry/14212
